contracts/utils/SafeCast.sol

```diff
@@ -0,0 +1,31 @@
+pragma solidity ^0.6.0;
+
+
+/// @dev Implements safe casting between int256 and uint256
+/// @author Alberto Cuesta Cañada
+library SafeCast {
+
+    /// @dev Maximum value that can be represented in an int256
+    function maxInt256() internal pure returns(int256) {
+        // solium-disable-next-line max-len
+        return 57896044618658097711785492504343953926634992332820282019728792003956564819967;
+    }
+
+    /// @dev Safe casting from int256 to uint256
+    function toUint(int256 x) internal pure returns(uint256) {
+        require(
+            x >= 0,
+            "Cannot cast negative signed integer to unsigned integer."
+        );
+        return uint256(x);
+    }
+
+    /// @dev Safe casting from uint256 to int256
+    function toInt(uint256 x) internal pure returns(int256) {
+        require(
+            x <= toUint(maxInt256()),
+            "Cannot cast overflowing unsigned integer to signed integer."
```

Comment:
Just a reminder to a 59 character revert message https://github.com/ethereum/solidity/issues/4588#issuecomment-408600802.
Remember that you can define error codes (eg. this would be `CAST3` which when looking at documentation would mean that error.). Remember that, when writing the UI, usually we do not cast the message itself. And if we do, we just want an identifier to understand what was the error and later on, present to the user. Such error in the UI would probably return to the user saying like "You possibly have an invalid input".
It's something we can consider. Do a PR just organize these error messages.

Reply:
We can continue this discussion at issue #290. I don't have an easy answer.